instructions/docs/add-the-encryption-sdk.md

```diff
@@ -0,0 +1,230 @@
+# Exercise 1: Add the AWS Encryption SDK
+
+## Background
+
+Right now, the Document Bucket supports storing files (or documents, or other blobs of data) in S3, and indexing them in DynamoDB. This allows Document Bucket users to share files with other users, or store them for retrieval later. The DynamoDB entries provide fast lookups to the content of the bucket, along with metadata context for each bucket item.
+
+This context allows storing additional information about the S3 item. Perhaps the origin user, the destination fleet, the project, or any other tag that would be useful to know without downloading and examining the item.
+
+DynamoDB is also configured to allow indexing on which documents have which keys. So, for example, it's a quick query to find out which documents in the bucket have been tagged with "configuration" as a piece of metadata about the object contents.
+
+Here's the API the Document Bucket supports:
+
+* `list`: This operation queries DynamoDB for all entries for all items in the bucket, and their metadata. It returns the `set` of items that have been stored.
+* `store`: This operation accepts a blob of bytes and a `map` of metadata context. It generates a unique identifier for the document. The identifier and associated metadata are written to DynamoDB. The bytes of the data are written to S3 under a key of that unique identifier. Any context metadata keys in DynamoDB are updated to include that new object identifier.
+* `retrieve`: This operation accepts a unique identifier as an argument. First it looks that identifier up in DynamoDB and pulls the identifier and its context out. Then it retrieves that object from S3. It bundles these items together and returns them to the caller.
+* `search`: This operation accepts a metadata key to search for. It then queries DynamoDB for the `set` of documents in the Document Bucket that have context matching that key. This operation then returns that set of document identifiers and their metadata.
+    * Once the desired document or documents have been identified with the returned metadata, the document identifiers can be passed to `retrieve` to actually fetch the documents.
+
+This is a start for sharing, storing, and searching documents of a variety of types. But what about sensitive documents? Or protecting, say, important configuration files from accidental corruption during storing or retrieving?
+
+Now you will add the AWS Encryption SDK to take an extra step of protection: client-side encrypting the Document Bucket document before it is transmitted off of the host machine. You will use KMS to provide a `data key` for each document, using a CMK that you set up in [Getting Started](./getting-started.md) when you deployed your stacks.
```

Comment:
instead of calling this "extra step of protection" could we say "to encrypt closer to where the data originates/is processed"

Reply:
can add a parenthetical clarifying that encrypting after it has been transmitted off the host machine it's now "server-side"?

Reply:
can we clarify that the 'data key' is actually an identical data key pair, one of which is encrypted under the CMK? I expect some number of people will attempt this guide without having attended the accompanying talk

Reply:
Yes. Ish. Yes although I was thinking of writing a background introduction to address that case instead.

Reply:
Having noodled on it a bit, I am going to defer the `data key` discussion to that background page, if that's all right?

Reply:
https://github.com/aws-samples/busy-engineers-document-bucket/issues/64